fastavro/_read.pyx

```diff
@@ -64,10 +70,16 @@ cpdef match_types(writer_type, reader_type):
         return True
     elif writer_type == "bytes" and reader_type == "string":
         return True
+    writer_schema = named_schemas["writer"].get(writer_type)
+    reader_schema = named_schemas["reader"].get(reader_type)
+    if writer_schema is not None and reader_schema is not None:
+        return match_types(writer_schema, reader_schema, named_schemas)
     return False
 
 
-cpdef match_schemas(w_schema, r_schema):
+cpdef match_schemas(w_schema, r_schema, named_schemas=None):
```

Comment:
I think named_schemas is always passed in, so this can just be a required arg instead of optional.